### PR TITLE
Deferred ListView.separator cleanups

### PR DESCRIPTION
**Pull request by HansMuller**
Updates for https://github.com/flutter/flutter/pull/18619 per https://github.com/flutter/flutter/pull/18619#issuecomment-399180832 and https://github.com/flutter/flutter/pull/18619#discussion_r197295024 .

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM after the CLA bot is happy.

File: packages/flutter/lib/src/widgets/scroll_view.dart [hunk 1]
```diff
@@ -452,21 +452,26 @@ abstract class BoxScrollView extends ScrollView {
 /// machinery can make use of the foreknowledge of the children's extent to save
 /// work, for example when the scroll position changes drastically.
 ///
-/// There are three options for constructing a [ListView]:
+/// There are four options for constructing a [ListView]:
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_view.dart]: (you don't have to make this change, but fwiw, this diff i why i usually say "There are several options...")

**Comment by Hixie** [comment]
I'm confident that Hans' employer has signed the CLA.

**Comment by Hixie** [comment]
This can land on green.

**Pull request merged**